intermediate_source/rpc_param_server_tutorial.rst

```diff
@@ -0,0 +1,398 @@
+       assert args.num_gpus <= 3, f"Only 0-2 GPUs currently supported (got {args.num_gpus})."
+       os.environ['MASTER_ADDR'] = args.master_addr
+       os.environ["MASTER_PORT"] = args.master_port
+
+Now, we'll create a process corresponding to either a parameter server or trainer depending on our command line arguments. We'll create a ``ParameterServer`` if our passed in rank is 0, and a ``TrainerNet`` otherwise. Note that we're using ``torch.multiprocessing`` to launch a subprocess corresponding to the function that we want to execute, and waiting on this process's completion from the main thread with ``p.join()``. In the case of initializing our trainers, we also use PyTorch's `dataloaders <https://pytorch.org/docs/stable/data.html>`_ in order to specify train and test data loaders on the MNIST dataset. 
+
+.. code-block:: python
+
+   processes = []
+       world_size = args.world_size
+       if args.rank == 0:
+           p = mp.Process(target=run_parameter_server, args=(0, world_size))
+           p.start()
+           processes.append(p)
+       else:
+           # Get data to train on
+           train_loader = torch.utils.data.DataLoader(
+               datasets.MNIST('../data', train=True, download=True,
+                              transform=transforms.Compose([
+                                  transforms.ToTensor(),
+                                  transforms.Normalize((0.1307,), (0.3081,))
+                              ])),
+               batch_size=32, shuffle=True,)
+           test_loader = torch.utils.data.DataLoader(
+               datasets.MNIST(
+                   '../data',
+                   train=False,
+                   transform=transforms.Compose(
+                       [
+                           transforms.ToTensor(),
+                           transforms.Normalize(
+                               (0.1307,
+                                ),
+                               (0.3081,
+                                ))])),
+               batch_size=32,
+               shuffle=True,
+           )
+           # start training worker on this node
+           p = mp.Process(
+               target=run_worker,
+               args=(
+                   args.rank,
+                   world_size, args.num_gpus,
+                   train_loader,
+                   test_loader))
+           p.start()
+           processes.append(p)
+
+       for p in processes:
+           p.join()
+
+To run the example locally, run the following command worker for the server and each worker you wish to spawn, in separate terminal windows: ``python rpc_parameter_server.py [world_size] [rank] [num_gpus]``. For example, for a master node with world size of 2, the command would be ``python rpc_parameter_server.py 2 0 0``. The trainer can then be launched with the command ``python rpc_parameter_server.py 2 1 0`` in a separate window, and this will begin training with one server and a single trainer.
```

Comment:
since we have arg parser, should we make the cmd line args explicit? e.g., `python rpc_parameter_server.py --world_size=2 --rank=0 --num_gpus=0`